data/src/main/java/com/linkedin/data/collections/CheckedMap.java

```diff
@@ -481,50 +482,64 @@ private boolean assertCheckMap(Map<? extends K, ? extends V> map)
 
   private void notifyChangeListenersOnPut(K key, V value)
   {
-    if (_changeListeners == null)
+    purgeStaleChangeListeners();
+    if (_changeListenerHead == null)
     {
       return;
     }
-
-    _changeListeners.forEach(listenerRef -> {
+    WeakListNode<ChangeListener<K, V>> node = _changeListenerHead;
```

Comment:
Suggest providing a utility iteration method to keep logic simple?

Reply:
good point, done.